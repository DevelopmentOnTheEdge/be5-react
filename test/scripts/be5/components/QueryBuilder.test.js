import React          from 'react';
import renderer       from 'react-test-renderer';
import QueryBuilder   from '../../../../src/scripts/be5/components/QueryBuilder';
import {shallow, mount, render} from 'enzyme';
import testUtils      from "../testUtils";
import { Provider }   from 'react-redux';
import testData       from '../testData.json'


test('snapshot', () => {
  const json = Object.assign({}, testData.simpleTable, {
    included: [{attributes: {
      finalSql: 'SELECT * FROM users',
      sql: 'select * from users'
    }}]
  });

  const component = renderer.create(
    <Provider store={testUtils.getStore()}>
      <QueryBuilder value={json}/>
    </Provider>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
