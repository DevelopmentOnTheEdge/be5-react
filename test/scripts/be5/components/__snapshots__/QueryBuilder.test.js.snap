// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="queryBuilder"
>
  <h1>
    Query Builder
  </h1>
  <div
    className="SplitPane  horizontal "
    style={
      Object {
        "MozUserSelect": "text",
        "WebkitUserSelect": "text",
        "bottom": 0,
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "height": "100%",
        "minHeight": "100%",
        "msUserSelect": "text",
        "outline": "none",
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "userSelect": "text",
        "width": "100%",
      }
    }
  >
    <div
      className="Pane horizontal Pane1  "
      style={
        Object {
          "display": "flex",
          "flex": "none",
          "height": 300,
          "outline": "none",
          "position": "relative",
        }
      }
    >
      <div
        id="UNIQUE_ID_OF_DIV"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
    <span
      className="Resizer horizontal "
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    />
    <div
      className="Pane horizontal Pane2  "
      style={
        Object {
          "flex": 1,
          "outline": "none",
          "position": "relative",
        }
      }
    >
      <div>
        <br />
        <div
          className="document-content"
          id="document-content___queryBuilder-table"
        >
          <div>
            <div
              className="table-component"
            >
              <h1
                className="table-component__title"
              >
                Организации: Общие сведения
              </h1>
              <div>
                <div
                  className="operationList"
                >
                  <button
                    className="btn btn-secondary btn-secondary-old btn-sm"
                    onClick={[Function]}
                  >
                    Редактировать
                  </button>
                   
                  <button
                    className="btn btn-secondary btn-secondary-old btn-sm"
                    onClick={[Function]}
                  >
                    Добавить
                  </button>
                </div>
                <div
                  id="quickColumns"
                >
                  <span>
                    Другие колонки:
                  </span>
                  <span>
                    <input
                      checked={false}
                      id="quick0"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <label
                      className="rowIndex"
                      htmlFor="quick0"
                    >
                      Наименование
                       
                    </label>
                  </span>
                </div>
                <div
                  className="scroll"
                >
                  <div
                    className="row"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <h2>
          Final sql
        </h2>
        <pre>
          SELECT * FROM users
        </pre>
      </div>
    </div>
  </div>
</div>
`;
